### DISCUSSION TITLE
Imagemagick trim and density

### DISCUSSION THREAD

**Discussion by Beccamira-create** [ImageMagick/ImageMagick#7557] [Help]
### ImageMagick version

7.1.1-Q8

### Operating system, version and so on

WIndows 11

### Description

Hello: I am trying to create a lopp to convert many images from .ps extension to  .jpg but it seems I am doing something wrong and I do not see how to fix it. Can someone give me a hand?
 Thank you.

magick_exists="TRUE"

if [ ${magick_exists} == "TRUE" ] ; then

if [ -a plot_200.jpg ] ; then rm plot_200.jpg ; fi
if [ -a plot_200.ps ] ; then
magick -trim -density 150 plot_200.ps plot_200.jpg ; fi

fi

**Comment by snibgo** [comment]
What error messages do you get? Please copy them and paste them here.

**Comment by Beccamira-create** [comment]
appears a message: Invalid Parameter - -density  

and I do not get the file in .jpg

**Comment by fmw42** [comment]
> magick -trim -density 150 plot_200.ps plot_200.jpg

You cannot trim before you have read the input.  Try

`magick -density 150 plot_200.ps -trim plot_200.jpg`

**Reply by Beccamira-create** [reply]
Ah thank you, let me try.

Now it says: Invalid Parameter - 150

**Comment by snibgo** [comment]
fmw42 is correct, but there is a more fundamental problem.

"invalid parameter" most likely comes from a Microsoft program, not magick.

You have shown a bash script. I suppose you have bash on your Windows computer? How are you running that script?

You say:

> I am trying to create a lopp to convert many images ...

... but that script has no loop.

Please show the text output from `magick -version`.

Then try the following commands:

```
magick rose: r.png

magick r.png r.jpg
```
Has this written the files `r.png` and  `r.jpg`?

**Reply by Beccamira-create** [reply]
Yes I am running bash. 

When I run magick -version in the CMD, a message appear: 
Version: ImageMagick 7.1.1-36 Q8 x64 852a4e9:20240727 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (194033812)

**Comment by Beccamira-create** [comment]
When I run (in the CMD): 
magick rose: r.png
magick r.png r.jpg

Nothing appears.

**Comment by Beccamira-create** [comment]
Magick -version --> Command 'magic' not found, but can be installed with sudo apt install 

Then I wrote sudo apt install --> It seemed it installed a lot of packages in ubuntu and te last lines say: Setting up imagemagick (8:6.9.11.60+dfsg-1.3ubuntu0.22.04.5) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for libc-bin (2.35-0ubuntu3.8) ...

**Comment by Beccamira-create** [comment]
And when I run my script to conver the ps to jpg a message says:  magick: command not found

**Comment by Beccamira-create** [comment]
I coould somehow get the .jpg images, but I have problems getting the images in pdf
I wrote:  policy domain="coder" rights="none" pattern="{PS,PS2,PS3,EPS,PDF,XPS}" /
Result -->Command 'policy' not found, did you mean:
  command 'police' from snap police (0+git.9492c4b)
See 'snap info <snapname>' for additional versions.

then Iw rote: sudo sed -i_bak \
's/rights="none" pattern="PDF"/rights="read | write" pattern="PDF"/'  \
/etc/ImageMagick-6/policy.xml

policy domain="coder" rights="read | write" pattern="PDF" /

result -->Command 'policy' not found, did you mean:
  command 'police' from snap police (0+git.9492c4b)
See 'snap info <snapname>' for additional versions.

**Comment by snibgo** [comment]
What bash do you use?

I use Cygwin bash. From the Windows CMD command line, I can type `bash -version` and the text output is:
```
GNU bash, version 4.4.12(3)-release (x86_64-unknown-cygwin)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Cygwin bash calls the Windows operating system for the low-level work (file I/O, console I/O, etc). This means that programs that are built for Windows will work with no change whether called from either shell, Windows CMD or Cygwin bash. And programs built for Cygwin bash can also be called from either shell.

Other versions of bash work in other ways. For example, if I understand correctly, Microsoft "Windows Subsystem for Linux" (WSL) is a replacement operating system, and has different low-level calls. So programs built for Linux will work on WSL but not CMD. And programs built for CMD will work for CMD but not for WSL.

(I have never used WSL.)

I think you have installed ImageMagick for Windows CMD. If you use WSL, then I think you need to install ImageMagick for whatever Linux distribution you use.

**Comment by fmw42** [comment]
> I wrote: policy domain="coder" rights="none" pattern="{PS,PS2,PS3,EPS,PDF,XPS}"

rights="none" means no read or write.  So you want to change that to rights="read|write" so that you can either read or write to those formats.

**Comment by Beccamira-create** [comment]
policy domain="coder" rights="read|write" pattern="{PS,PS2,PS3,EPS,PDF,XPS,JPG}"

-->>Command 'policy' not found, did you mean:
  command 'police' from snap police (0+git.9492c4b)
See 'snap info <snapname>' for additional versions.

**Comment by snibgo** [comment]
@Beccamira-create: You are trying to run `policy` as a command, so you get errors like `Command 'policy' not found`. It isn't a command. It is an entry in the ImageMagick file `policy.xml`.
